drivers/adc/adc_stm32.c

```diff
@@ -307,11 +312,11 @@ static int adc_stm32_dma_start(const struct device *dev,
 	/* Source and destination */
 	blk_cfg->source_address = (uint32_t)LL_ADC_DMA_GetRegAddr(adc, LL_ADC_DMA_REG_REGULAR_DATA);
 	blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
-	blk_cfg->source_reload_en = 1;
+	blk_cfg->source_reload_en = 0;
 
 	blk_cfg->dest_address = (uint32_t)buffer;
 	blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_INCREMENT;
-	blk_cfg->dest_reload_en = 1;
+	blk_cfg->dest_reload_en = 0;
```

Comment:
This should be 1, as it needs to reload the destination buffer ptr to the start address again, although since we're not running it continuously it should not matter. But it's still better to do so.

Reply:
This is not possible, due to this check in the DMA driver.
```
	if (config->head_block->source_reload_en !=
		config->head_block->dest_reload_en) {
		LOG_ERR("source_reload_en and dest_reload_en must "
			"be the same.");
		return -EINVAL;
	}
```